src/Webserver.cc

```diff
@@ -195,3 +194,20 @@ std::string Webserver::buffer_to_string(const boost::asio::streambuf &buffer)
   std::string result(buffers_begin(bufs), buffers_begin(bufs) + buffer.size());
   return result;
 }
+
+std::string Webserver::find_prefix(std::string uri) {
+    std::string longest;
+    size_t last = uri.find_last_of("/");
+    std::string prefix = uri.substr(0, last);
+
+    for (auto it : handler_map) {
+        std::string map = it.first;
+        if (prefix.find(map) == 0) {
```

Comment:
I was testing something, and I found out this doesn't work, because you have to check in chunks, not just a pure prefix. For example, in my config I had /s as the mapping, but when I tried to access /static/file.txt, it went through because /s is a prefix of /static. We have to check the prefix in chunks delimited by slashes I think